packages/flutter_tools/bin/xcode_backend.sh

```diff
@@ -46,37 +46,41 @@ BuildApp() {
     target_path="${FLUTTER_TARGET}"
   fi
 
+  # Use FLUTTER_BUILD_MODE if it's set, otherwise use the Xcode build configuration name
+  # This means that if someone wants to use an Xcode build config other than Debug/Profile/Release,
+  # they _must_ set FLUTTER_BUILD_MODE so we know what type of artifact to build.
+  local build_mode="$(echo "${FLUTTER_BUILD_MODE:-${CONFIGURATION}}" | tr "[:upper:]" "[:lower:]")"
+  local artifact_variant="unknown"
+  case "$build_mode" in
+    release) artifact_variant="ios-release";;
+    profile) artifact_variant="ios-profile";;
+    debug) artifact_variant="ios";;
+    *) 
+      EchoError "========================================================================"
+      EchoError "ERROR: Unknown FLUTTER_BUILD_MODE: ${build_mode}."
+      EchoError "Valid values are 'Debug', 'Profile', or 'Release' (case insensitive)."
+      EchoError "This is controlled by the FLUTTER_BUILD_MODE environment varaible."
+      EchoError "If that is not set, the CONFIGURATION environment variable is used."
+      EchoError "========================================================================"
+      exit -1;;
+  esac
+
   # Archive builds (ACTION=install) should always run in release mode.
-  if [[ "$ACTION" == "install" && "$FLUTTER_BUILD_MODE" != "release" ]]; then
+  if [[ "$ACTION" == "install" && "$build_mode" != "release" ]]; then
     EchoError "========================================================================"
     EchoError "ERROR: Flutter archive builds must be run in Release mode."
     EchoError ""
-    EchoError "To correct, run:"
+    EchoError "To correct, ensure FLUTTER_BUILD_MODE to release or run:"
```

Comment:
"ensure"?

Reply:
oh maybe you meant ensure .. is set?

Reply:
Should be `ensure FLUTTER_BUILD_MODE is set to release`...  are you saying it should use a verb other than ensure?